keps/sig-node/5474-enable-writable-cgroups/README.md

```diff
@@ -0,0 +1,505 @@
+
+#### Story 2: Dynamic Resource Management
+
+As an application developer building a resource-intensive application, I want my application to be able to monitor its memory usage via `/sys/fs/cgroup/memory.current` and dynamically adjust its memory limits via `/sys/fs/cgroup/memory.max` based on workload patterns, without requiring privileged access or manual operator intervention.
+
+### Notes/Constraints/Caveats (Optional)
+
+- **cgroup v2 Only**: This feature requires cgroup v2 and will be ignored on cgroup v1 systems
+- **Linux Only**: The field is only valid on Linux containers and will be validated accordingly
+- **Runtime Support**: Requires container runtime support (Containerd, CRI-O)
+- **Security Context Integration**: Must work cohesively with other SecurityContext fields
+
+### Risks and Mitigations
+
+| Risk | Mitigation |
+|------|------------|
+| **Security Bypass**: Containers gaining unauthorized access to system cgroups | Only allow write access to container's own cgroup subtree. cgroup v2 delegation model provides isolation |
+| **Resource Exhaustion**: Containers setting inappropriate resource limits | Kubernetes resource quotas and limit ranges still apply. Container cannot exceed pod-level limits |
+| **Pod Security Policy Bypass**: Feature being used in restricted environments | Integration with Pod Security Standards to block in restricted profiles |
+| **Intra-Pod Resource Starvation**: A container in a Burstable or BestEffort Pod could modify its cgroup limits to consume resources intended for other containers in the same Pod, as no Pod-level cgroup enforces a ceiling. | Enforce that any container setting WritableCgroups: true must belong to a Pod that qualifies for the Guaranteed QoS class (where all containers have CPU and memory requests and limits set, and they are equal). This can be done via a validating admission policy or potentially in the normal validation code. This ensures Kubelet creates a pod-level cgroup that acts as a hard boundary for all containers within the pod. | 
+| **Runtime Incompatibility**: Feature not working with older runtimes | Graceful degradation - field ignored if runtime doesn't support it |
+
+## Design Details
+
+### API Changes
+
+#### Core API Types
+
+Add `WritableCgroups *bool` field to `SecurityContext` in both internal and external API:
+
+**File**: `pkg/apis/core/types.go`
+```go
+type SecurityContext struct {
+    // ... existing fields ...
+    
+    // WritableCgroups controls whether the container has write access to cgroup interfaces.
+    // This allows unprivileged containers to manage their own cgroup hierarchies on cgroup v2 systems.
+    // Only effective on Linux containers with cgroup v2.
+    // +optional
+    WritableCgroups *bool
+}
+```
```

Comment:
https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/core#NodeRuntimeHandlerFeatures should be updated too